src/application/look-up-wallet-address/address-list.tsx

```diff
@@ -0,0 +1,103 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { classes, Icon } from 'shared/ui';
+interface AddressListProperties {
+  lookupText: string | undefined;
+  foundAddresses: Record<string, string> | undefined;
+  isTwitterLookup: boolean;
+  onAddressCopied: () => void;
+}
+export const AddressList = ({
+  foundAddresses,
+  isTwitterLookup,
+  lookupText,
+  onAddressCopied,
+}: AddressListProperties) => {
+  const [copiedAddressValue, setCopiedAddressValue] = useState<string>();
+
+  const handleAddressClick = useCallback((address: string) => {
+    void navigator.clipboard.writeText(address);
+    setCopiedAddressValue(address);
+  }, []);
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout;
+    if (copiedAddressValue) {
+      timeout = setTimeout(() => {
+        setCopiedAddressValue(undefined);
+        onAddressCopied();
+      }, 750);
+    }
+    return () => {
+      if (timeout) {
+        clearTimeout(timeout);
+      }
+    };
+  }, [copiedAddressValue, onAddressCopied]);
```

Comment:
I asked GPT and seems like it can be done without useEffect:
```tsx
const handleAddressClick = useCallback((address: string) => {
  void navigator.clipboard.writeText(address);

  const timeout = setTimeout(() => {
    onAddressCopied();
  }, 750);

  // Clean up the timeout if this function is called again
  return () => {
    clearTimeout(timeout);
  };
}, [onAddressCopied]);
```

The only thing left which you will have `copiedAddress` state for is that green colour after copying, we already decided that UX/UI is not very important here so maybe we can get rid of it and remove state completely

Reply:
I've decided to use the useEffect to allow copying other addresses in case of a misclick.
https://github.com/user-attachments/assets/b972345e-5949-465d-8bf9-27bf4b483bed

Reply:
Do you think it can cause any problems?